scio-neo4j/src/it/scala/com/spotify/scio/neo4j/Neo4jIOIT.scala

```diff
@@ -0,0 +1,117 @@
+    val session = client.session()
+    try {
+      session.writeTransaction { tx =>
+        tx.run(
+          """CREATE
+            | (usa:Country {name: 'USA'}),
+            | (charlie:Person {name: 'Charlie Sheen'}),
+            | (oliver:Person {name: 'Oliver Stone'}),
+            | (michael:Person {name: 'Michael Douglas'}),
+            | (martin:Person {name: 'Martin Sheen'}),
+            | (morgan:Person {name: 'Morgan Freeman'}),
+            | (wallStreet:Movie {name: 'Wall Street', year: 1987}),
+            | (americanPresident:Movie {name: 'American President', year: 1995}),
+            | (theShawshankRedemption:Movie {name: 'The Shawshank Redemption', year: 1994}),
+            | (charlie)-[:ACTED_IN {role: 'Bud Fox'}]->(wallStreet),
+            | (martin)-[:ACTED_IN {role: 'Carl Fox'}]->(wallStreet),
+            | (michael)-[:ACTED_IN {role: 'Gordon Gekko'}]->(wallStreet),
+            | (martin)-[:ACTED_IN {role: 'A.J. MacInerney'}]->(americanPresident),
+            | (michael)-[:ACTED_IN {role: 'President Andrew Shepherd'}]->(americanPresident),
+            | (morgan)-[:ACTED_IN {role: 'Ellis Boyd \'Red\' Redding'}]->(theShawshankRedemption)
+            |""".stripMargin
+        ).consume()
+        ()
+      }
+    } finally session.close()
+
+    val options = PipelineOptionsFactory.create()
+    options.setRunner(classOf[DirectRunner])
+
+    val martin = Person("Martin Sheen")
+    val expectedRoles = Seq(
+      Role(martin, Movie("Wall Street", Year.of(1987)), "Carl Fox"),
+      Role(martin, Movie("American President", Year.of(1995)), "A.J. MacInerney")
+    )
+
+    val neo4jOptions = Neo4jReadOptions(
+      Neo4jConnectionOptions(container.boltUrl, container.username, container.password),
+      s"MATCH (p)-[r: ACTED_IN]->(m) WHERE p.name='${martin.name}' RETURN p, r, m",
+      row => {
+        val p = Person(row.get("p").get("name").asString())
+        val m =
+          Movie(row.get("m").get("name").asString(), Year.of(row.get("m").get("year").asInt()))
+        Role(p, m, row.get("r").get("role").asString())
+      }
+    )
+
+    runWithRealContext(options) { sc =>
+      sc.neo4jCypher(neo4jOptions) should containInAnyOrder(expectedRoles)
+    }
+  }
+
+  it should "write to the graph database" in {
+    val options = PipelineOptionsFactory.create()
+    options.setRunner(classOf[DirectRunner])
+
+    val movieNames = Seq("Wall Street", "American President", "The Shawshank Redemption")
+
+    val neo4jOptions = Neo4jWriteOptions[(String, String)](
+      Neo4jConnectionOptions(container.boltUrl, container.username, container.password),
+      """UNWIND $origin AS origin
+        |MATCH
+        |  (m:Movie {name: origin.movie}),
+        |  (c:Country {name: origin.country})
+        |CREATE (m)-[:ORIGIN]->(c)
+        |""".stripMargin,
+      "origin",
+      { case (movie, country) => Map("movie" -> movie, "country" -> country) }
```

Comment:
Same thing here
```scala
.saveAsNeo4j(neo4jOptions, cypher)(x => Map(...))
```
As a neo4j newbie, I'm also confused by the requirement of the `UNWIND` and the obligation to declare the `unwindMapName`
As I understand, this is to enable batching.

Can this be hidden to the user ?
```
MATCH
  (m:Movie {name: $movie}),
  (c:Country {name: $country})
CREATE (m)-[:ORIGIN]->(c)
```
could be automatically transformed with a carefully picked unwinded name, replacing the `$`

Reply:
> As a neo4j newbie, I'm also confused by the requirement of the UNWIND and the obligation to declare the unwindMapName
> As I understand, this is to enable batching.
> 
> Can this be hidden to the user ?

Maybe so. however, I hereby notify — bother — @mattcasters again, since he is the best expert we can find here, I believe!

Reply:
Just to clarify the UNWIND situation: I should mention that without UNWIND you really can't get decent performance with Neo4j.  It is also painful to note that the dialect for UNWIND is a bit different from regular Cypher statements. However, I'm told this is something my colleagues in engineering are working on for the upcoming version 5 of Neo4j.  I propose to wait until that situation becomes a bit clearer, before making additional changes. Fingers crossed.